sdk/samples/iot/hub/src/paho_iot_hub_c2d_example.c

```diff
@@ -149,35 +188,62 @@ static az_result read_configuration_and_init_client()
   return AZ_OK;
 }
 
-static int on_received(void* context, char* topicName, int topicLen, MQTTClient_message* message)
+// Read OS environment variables using stdlib function
+static az_result read_configuration_entry(
+    const char* name,
+    const char* env_name,
+    char* default_value,
+    bool hide_value,
+    az_span buffer,
+    az_span* out_value)
 {
-  (void)context;
+  printf("%s = ", name);
+  char* env = getenv(env_name);
 
-  if (topicLen == 0)
+  if (env != NULL)
   {
-    // The length of the topic if there are one or more NULL characters embedded in topicName,
-    // otherwise topicLen is 0.
-    topicLen = (int)strlen(topicName);
+    printf("%s\n", hide_value ? "***" : env);
+    az_span env_span = az_span_from_str(env);
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(env_span));
+    az_span_copy(buffer, env_span);
+    *out_value = az_span_slice(buffer, 0, az_span_size(env_span));
+  }
+  else if (default_value != NULL)
+  {
+    printf("%s\n", default_value);
+    az_span default_span = az_span_from_str(default_value);
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(default_span));
+    az_span_copy(buffer, default_span);
+    *out_value = az_span_slice(buffer, 0, az_span_size(default_span));
+  }
+  else
+  {
+    printf("(missing) Please set the %s environment variable.\n", env_name);
+    return AZ_ERROR_ARG;
   }
 
-  // Parse the incoming message topic and check to make sure it is a c2d message
-  az_iot_hub_client_c2d_request c2d_request;
-  if (az_succeeded(az_iot_hub_client_c2d_parse_received_topic(
-          &client, az_span_init((uint8_t*)topicName, topicLen), &c2d_request)))
+  return AZ_OK;
+}
+
+// Create mqtt endpoint e.g: ssl//contoso.azure-devices.net:8883
+static az_result create_mqtt_endpoint(char* destination, int32_t destination_size, az_span iot_hub)
```

Comment:
hummm?

```suggestion
static az_result get_mqtt_endpoint(char* destination, int32_t destination_size, az_span iot_hub)
```

Reply:
I view this as creating more than getting, in that its constructing the endpoint rather than retrieving something stored in memory.